File: app/src/main/java/com/amaze/filemanager/filesystem/ftpserver/AndroidFileSystemFactory.java
/*
 * AndroidFileSystemFactory.java
 *
 * Copyright © 2018 Raymond Lai <airwave209gt at gmail.com>.
 *
 * This file is part of AmazeFileManager.
 *
 * AmazeFileManager is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * AmazeFileManager is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with AmazeFileManager. If not, see <http ://www.gnu.org/licenses/>.
 */
package com.amaze.filemanager.filesystem.ftpserver;

import android.content.Context;
import android.support.annotation.NonNull;

import org.apache.ftpserver.ftplet.FileSystemFactory;
import org.apache.ftpserver.ftplet.FileSystemView;
import org.apache.ftpserver.ftplet.User;

public class AndroidFileSystemFactory implements FileSystemFactory {

    private final Context context;

    private final String ftpServerRoot;

    public AndroidFileSystemFactory(@NonNull final Context context, @NonNull String ftpServerRoot){
        this.context = context;
        this.ftpServerRoot = ftpServerRoot;
    }

    @Override
    public FileSystemView createFileSystemView(User user) {
        return new AndroidFileSystemView(context, ftpServerRoot);
    }
}

File: app/src/main/java/com/amaze/filemanager/filesystem/ftpserver/AndroidFileSystemFactory.kt
/*
 * Copyright (C) 2014-2021 Arpit Khurana <arpitkh96@gmail.com>, Vishal Nehra <vishalmeham2@gmail.com>,
 * Emmanuel Messulam<emmanuelbendavid@gmail.com>, Raymond Lai <airwave209gt at gmail.com> and Contributors.
 *
 * This file is part of Amaze File Manager.
 *
 * Amaze File Manager is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

package com.amaze.filemanager.filesystem.ftpserver

import android.content.Context
import android.os.Build.VERSION_CODES.KITKAT
import androidx.annotation.RequiresApi
import com.amaze.filemanager.asynchronous.services.ftp.FtpService
import org.apache.ftpserver.ftplet.FileSystemFactory
import org.apache.ftpserver.ftplet.FileSystemView
import org.apache.ftpserver.ftplet.User

@RequiresApi(KITKAT)
class AndroidFileSystemFactory(private val context: Context) : FileSystemFactory {

    override fun createFileSystemView(user: User?): FileSystemView =
        AndroidFtpFileSystemView(context, user?.homeDirectory ?: FtpService.defaultPath(context))
}
